documentation/user-consent.md

```diff
@@ -48,122 +43,270 @@ The document details aspects regarding the consent management, which includes fo
+
+The API exposure platform receives the request from the device application (Step 3) and does the following:
+
+- Use network based authentication mechanism to obtain the user identifier, i.e.: MSISDN. Set the OAuth sub to the unique user ID in the operator (Step 4).
+
+- Check if user consent is required, which depends on the legal basis associated with the purpose ("legitimate interest", "contract", "consent", etc). If necessary, it will check in the operator's consent master whether user consent has already been given for this identifier, the application client_id and the requested purpose (Steps 5-6).
+
+Then, two alternatives may occur:
+
+**Scenario 1**: User consent is not required or consent is already given (Step 7). The API exposure platform will continue the authorization code flow by redirecting to the API invoker redirect_uri (invoker_callback) and including the authorization code (OperatorCode).
+
+**Scenario 2**: Consent is required and not yet provided by user (Step 8)
+
+- The operator performs the consent capture. Since the authorization code grant involves the frontend, the consent can be captured directly from the user.
+- Once the user has given consent, the authorization code flow continues by redirecting to the API invoker redirect_uri (invoker_callback) and including the authorization code (OperatorCode).
+
+Once the API invoker receives the redirect with the authorization code (OperatorCode - Step 9), it will retrieve the access token from the operator's API exposure platform (OperatorAccessToken) (Steps 10-11). The OperatorAccessToken issued is encrypted so that no relevant information is exposed.
+
+Now the API invoker has a valid access token that can be used to invoke the CAMARA API offered by the operator (Step 12).
 
->[TO BE EDITED/COMPLETED] - Flow to obtain an access token to consume network API...
+The operator's API exposure platform will validate OperatorAccessToken, grant the access to the API based on the scopes bound to the access token, progress request to the corresponding API backend and retrieve the API response (Step 13).
+
+Finally, the operator will provide API response to the API invoker (Step 14).
+
+**Technical ruleset for the Frontend flow**
+
+On-net scenarios where the mobile connection of the device needs to be authenticated (e.g. This flow is for example the one specified for the [CAMARA Number Verification API](https://github.com/camaraproject/NumberVerification/blob/main/documentation/API_documentation/CAMARA/uml_v0.3.jpg) due to the nature of its functionality where a given MSISDN needs to be compared to the MSISDN associated with the mobile connection of the user device). The device application (front-end) must be able to handle browser redirects.
```

Comment:
We had a similar ruleset in mind but phrased slightly differently:
1. If all API usecases point to the need of On-net scneario and where the consumption device and authentication device are the same, the front end flow must be used. eg. NumberVerification
2. If some usecase/s for an API point to off-net scenarios and where consumption and authenticaiton devices could be different, the backend flow must be used.

Reply:
Hopefully fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/54/commits/c89cbf94fd92b39514e0974ab85638c3034023e5